tests/linearizability/watch.go

```diff
@@ -104,6 +105,9 @@ func validateMemberWatchResponses(t *testing.T, responses []watchResponse, expec
 		}
 
 		for _, event := range resp.Events {
+			if event.Kv.ModRevision <= lastProgressNotifyRevision {
+				t.Errorf("Got event with revision lower than last progress notify event, eventRevision: %d, progressNotifyRevision: %d", event.Kv.ModRevision, lastProgressNotifyRevision)
+			}
```

Comment:
This checks looks good to me. But we'd better add comment something like below to explain why we have such check. 

```
etcd will not send progress notification to a watcher until it has synced all events.
```

Reply:
Sure, let me put it in the code that is easier to access when debugging a test failure.